spec-main/api-app-path-spec.ts

```diff
@@ -0,0 +1,315 @@
+        fs.existsSync(app.getPath('exe')),
+        fs.existsSync(app.getPath('home')),
+        fs.existsSync(app.getPath('temp'))
+      ];
+      expect(paths).to.deep.equal([true, true, true]);
+    });
+
+    it('throws an error when the name is invalid', () => {
+      expect(() => {
+        app.getPath('does-not-exist' as any);
+      }).to.throw(/Failed to get 'does-not-exist' path/);
+    });
+
+    it('returns the overridden path', () => {
+      app.setPath('music', __dirname);
+      expect(app.getPath('music')).to.equal(__dirname);
+    });
+
+    if (process.platform === 'win32') {
+      it('gets the folder for recent files', () => {
+        const recent = app.getPath('recent');
+
+        // We expect that one of our test machines have overriden this
+        // to be something crazy, it'll always include the word "Recent"
+        // unless people have been registry-hacking like crazy
+        expect(recent).to.include('Recent');
+      });
+
+      it('can override the recent files path', () => {
+        app.setPath('recent', 'C:\\fake-path');
+        expect(app.getPath('recent')).to.equal('C:\\fake-path');
+      });
+    }
+  });
+
+  describe('setPath(name, path)', () => {
+    it('throws when a relative path is passed', () => {
+      const badPath = 'hey/hi/hello';
+
+      expect(() => {
+        app.setPath('music', badPath);
+      }).to.throw(/Path must be absolute/);
+    });
+
+    it('does not create a new directory by default', () => {
+      const badPath = path.join(__dirname, 'music');
+
+      expect(fs.existsSync(badPath)).to.be.false();
+      app.setPath('music', badPath);
+      expect(fs.existsSync(badPath)).to.be.false();
+
+      expect(() => { app.getPath(badPath as any); }).to.throw();
+    });
+  });
+
+  describe('setAppLogsPath(path)', () => {
+    const appLogsPath = path.join(realtmpdir, 'mylogs');
+    it('by default', async () => {
+      const output = await runTestApp('app-custom-path');
+      switch (process.platform) {
+        case 'darwin':
+          expect(output.appLogs).to.equal(path.join(os.homedir(), 'Library', 'Logs', 'Electron'));
+          break;
+        case 'win32':
+        default:
+          expect(output.appLogs).to.equal(path.join(output.appData, defaultAppName, 'logs'));
+          expect(output.appLogs).to.equal(path.join(output.userData, 'logs'));
+          break;
+      }
+    });
+
+    // Linux
+    // app path module setAppLogsPath() setAppLogsPath(/tmp/mylogs) - setAppLogsPath(/tmp/mylogs)
+    // /home/builduser/project/src/electron/spec-main/api-app-path-spec.ts
+    // AssertionError: expected '/home/builduser/.config/app-custom-path/logs' to equal '/tmp/mylogs'
+
+    // if (process.platform === 'win32') {
+    //   it(`setAppLogsPath('${appLogsPath}')`, async () => {
+    //     const output = await runTestApp('app-custom-path', `-custom-applogs=${appLogsPath}`)
+    //     expect(output.appLogs).to.equal(appLogsPath)
+    //   })
+    // }
```

Comment:
These appear to be new tests but are commented out. Am I reading that right, and if so, what is your intent for these?

Reply:
intermediate  tests, not needed, I removed them